tools/operator/launcher/k8s/network.go

```diff
@@ -2,241 +2,522 @@ package k8s
+				if nsConfig.K8s.DataPersistence == "true" || nsConfig.K8s.DataPersistence == "local" {
+					volumeMount := v12.VolumeMount{MountPath: "/opt/couchdb/data", Name: "data-storage"}
+					container.VolumeMounts = append(container.VolumeMounts, volumeMount)
+				}
+				container.Image = fmt.Sprintf("hyperledger/fabric-couchdb")
+				containers = append(containers, container)
+			}
+			l.Containers = containers
+			l.Volumes = k8s.volumesList("peer", org.Name, fmt.Sprintf("peer%d-%s", j, org.Name), nsConfig.K8s.DataPersistence, nsConfig.EnableNodeOUs)
+			l.Ports = []int32{peerPort, peerMetricsPort}
+			launchConfig = append(launchConfig, l)
+			peerPort++
+			peerMetricsPort++
+		}
+		for m := 0; m < org.NumCA; m++ {
+			l := k8s.caLaunch(m, org.Name, caImage)
+			l.Ports = []int32{caPort}
+			launchConfig = append(launchConfig, l)
+			caPort++
+		}
 	}
-	for i := 0; i < len(k.Arguments); i++ {
-		switch k.Action {
-		case "apply", "delete":
-			args = append(args, []string{"-f", k.Arguments[i]}...)
-		default:
-			args = append(args, k.Arguments[i])
+
+	var ordererPort int32 = 30000
+	var ordererMetricsPort int32 = 32500
+	for i := 0; i < len(nsConfig.OrdererOrganizations); i++ {
+		org := nsConfig.OrdererOrganizations[i]
+		for j := 0; j < org.NumOrderers; j++ {
+			err := k8s.generateOrdererConfig(fmt.Sprintf("orderer%d-%s", j, org.Name), org.Name, org.MSPID, ordererPort, ordererMetricsPort, ordererConfig)
+			if err != nil {
+				return nil, errors.Wrap(err, "failed to generate orderer configuration file")
+			}
+
+			l := LaunchConfig{Name: fmt.Sprintf("orderer%d-%s", j, org.Name), Type: "orderer"}
+			containers := make([]v12.Container, 0)
+			container := v12.Container{Name: "orderer", Command: []string{"orderer"}, Resources: k8s.resources("orderer", nsConfig.K8s.Resources.Orderers)}
+			container.Image = ordererImage
+			env := v12.EnvVar{Name: "FABRIC_LOGGING_SPEC", Value: nsConfig.OrdererFabricLoggingSpec}
+			container.Env = append(container.Env, env)
+			container.VolumeMounts = k8s.volumeMountLists("orderer", nsConfig.K8s.DataPersistence, nsConfig.EnableNodeOUs)
+			containers = append(containers, container)
+			l.Containers = containers
+			l.Volumes = k8s.volumesList("orderer", org.Name, fmt.Sprintf("orderer%d-%s", j, org.Name), nsConfig.K8s.DataPersistence, nsConfig.EnableNodeOUs)
+			l.Ports = []int32{ordererPort, ordererMetricsPort}
+			launchConfig = append(launchConfig, l)
+			ordererPort++
+			ordererMetricsPort++
+		}
+		for m := 0; m < org.NumCA; m++ {
+			l := k8s.caLaunch(m, org.Name, caImage)
+			l.Ports = []int32{caPort}
+			launchConfig = append(launchConfig, l)
+			caPort++
 		}
 	}
-	return args
+	return launchConfig, nil
 }
 
-func (k K8s) ConfigMapsNSecretsArgs(componentName, k8sType string) []string {
+func (k8s K8s) resources(component string, resource networkspec.Resource) v12.ResourceRequirements {
```

Comment:
The `component` parameter is unused, we can remove it.

Reply:
thank you for identifying this. I overlooked it. Fixed.